app/lib/features/news/pages/add_news_page.dart

```diff
@@ -405,6 +374,17 @@ class AddNewsState extends ConsumerState<AddNewsPage> {
               slidePost.backgroundColor?.value,
             ),
           );
+
+          if (slidePost.newsReferencesModel != null) {
+            final linkBuilder = sdk.api.newLinkRefBuilder(
+              slidePost.newsReferencesModel!.type.name,
+              slidePost.newsReferencesModel!.id ?? '',
+            );
+            final linkRef = linkBuilder.build();
+            final objBuilder = sdk.api.newObjRefBuilder(null, linkRef);
+            final objRef = objBuilder.build();
```

Comment:
looks being done repetitively, maybe we want to put that into a separate function and just call that?

Reply:
Yupp! let me manage it in better way to avoid duplication of the code.